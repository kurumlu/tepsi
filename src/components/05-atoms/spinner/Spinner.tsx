import * as React from 'react';
import { FunctionComponent } from 'react';
import cn from 'classnames';

export enum SpinnerSize {
  LARGE = 'large', // 48px
  SMALL = 'small', // 24px
}

export type SpinnerProps = {
  size?: SpinnerSize;
};

export const Spinner: FunctionComponent<SpinnerProps> = ({
  size = SpinnerSize.LARGE,
}) => (
  <svg
    className={cn('sd-a-spinner', `sd-a-spinner--${size}`)}
    viewBox="0 0 40 40"
    xmlns="http://www.w3.org/2000/svg">
    <circle
      className="sd-a-spinner__ring"
      fill="none"
      strokeLinecap="round"
      strokeWidth="3"
      cx="20"
      cy="20"
      r="18"></circle>
  </svg>
);
