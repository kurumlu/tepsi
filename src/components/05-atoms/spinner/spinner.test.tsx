import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';

import { Spinner, SpinnerSize, SpinnerProps } from './Spinner';

describe('renders <Spinner />', () => {
  const component: ShallowWrapper<SpinnerProps> = shallow(<Spinner />);

  it('renders <Spinner /> component', () => {
    expect(component).toMatchSnapshot('default');
  });

  it('has specific class dependant on size', () => {
    const SmallSpinner: ShallowWrapper<SpinnerProps> = shallow(
      <Spinner size={SpinnerSize.SMALL} />
    );
    expect(SmallSpinner).toMatchSnapshot('small');
    expect(
      SmallSpinner.hasClass(`as-a-spinner--${SpinnerSize.SMALL}`)
    ).toBeTruthy();
  });
});
