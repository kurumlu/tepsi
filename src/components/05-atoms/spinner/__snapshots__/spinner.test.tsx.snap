// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders <Spinner /> has specific class dependant on size: small 1`] = `
<svg
  className="as-a-spinner as-a-spinner--small"
  viewBox="0 0 40 40"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    className="as-a-spinner__ring"
    cx="20"
    cy="20"
    fill="none"
    r="18"
    strokeLinecap="round"
    strokeWidth="3"
  />
</svg>
`;

exports[`renders <Spinner /> renders <Spinner /> component: default 1`] = `
<svg
  className="as-a-spinner as-a-spinner--large"
  viewBox="0 0 40 40"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    className="as-a-spinner__ring"
    cx="20"
    cy="20"
    fill="none"
    r="18"
    strokeLinecap="round"
    strokeWidth="3"
  />
</svg>
`;
